instrumentation/rails/lib/opentelemetry/instrumentation/rails/middlewares/tracer_middleware.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Rails
+      module Middlewares
+        # TracerMiddleware propagates context and instruments Rails requests
+        # by way of its middleware system
+        class TracerMiddleware
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            extracted_context = OpenTelemetry.propagation.http.extract(env)
+            OpenTelemetry::Context.with_current(extracted_context) do
+              tracer.in_span(env['PATH_INFO'], attributes: request_span_attributes(env: env), kind: :server) do |request_span|
+                @app.call(env).tap do |status, headers, response|
+                  set_attributes_after_request(
```

Comment:
How does this work with exceptions? Even in the presence of exceptions, we should try to set some of the attributes (we just need to be more defensive than usual).